tests/aio/test_tx.py

```diff
@@ -49,3 +49,45 @@ async def test_credentials():
         raised = True
 
     assert raised
+
+
+@pytest.mark.asyncio
+async def test_tx_snapshot_ro(driver, database):
+    session = await driver.table_client.session().create()
+    description = (
+        ydb.TableDescription()
+        .with_primary_keys("key")
+        .with_columns(
+            ydb.Column("key", ydb.OptionalType(ydb.PrimitiveType.Uint64)),
+            ydb.Column("value", ydb.OptionalType(ydb.PrimitiveType.Uint64)),
+        )
+    )
+    await session.create_table(database + "/test", description)
+    await session.transaction(ydb.SerializableReadWrite()).execute(
+        """INSERT INTO `test` (`key`, `value`) VALUES (1, 1), (2, 2)""",
+        commit_tx=True,
+    )
+
+    ro_tx = session.transaction(tx_mode=ydb.SnapshotReadOnly())
+    data1 = await ro_tx.execute("SELECT value FROM `test` WHERE key = 1")
+
+    rw_tx = session.transaction(ydb.SerializableReadWrite())
+    await rw_tx.execute("UPDATE `test` SET value = value + 1")
+    data2 = await ro_tx.execute("SELECT value FROM `test` WHERE key = 1")
```

Comment:
no need data2: changes from rw_tx not visible for any other transactions before commit rw_tx.